webSites/webSites-war/src/main/java/org/silverpeas/components/websites/servlets/WebSitesRequestRouter.java

```diff
@@ -90,1056 +96,776 @@ public WebSiteSessionController createComponentSessionController(
+      final HttpRequest request) throws WebSitesException {
+    String id = request.getParameter("Id");
+    // = null ou rempli si type= design
+    String currentPath = request.getParameter("path");
+    String type = request.getParameter("type"); // null ou design
+
+    request.setAttribute("Site", scc.getWebSite(id));
+    request.setAttribute("AllIcons", scc.getAllIcons());
+    request.setAttribute("ListIcons", scc.getIcons(id));
+
+    String recupParam = request.getParameter("RecupParam"); // =null ou oui
+    String complete = null;
+    if (recupParam != null) {// =oui
+      String nom = request.getParameter("Nom");
+      String description = request.getParameter(DESCRIPTION_PARAM);
+      String lapage = request.getParameter("Page");
+      String listeIcones = request.getParameter(ICON_LIST_PARAM);
+
+      type = "design";
+      complete =
+          "&RecupParam=oui&Nom=" + nom + "&Description=" + description + "&Page=" + lapage +
+              "&ListeIcones=" + listeIcones;
+    } else {
+      return "/webSites/jsp/modifDesc.jsp?Id=" + id + FOLDER_PATH_FILTER + currentPath + "&type=" +
+          type;
+    }
 
-          SiteDetail ancien = scc.getSessionSite();
-          id = ancien.getSitePK().getId();
-          int type = ancien.getType();
-
-          /* verif que le nom de la page principale est correcte */
-          Collection<File> collPages = scc.getAllWebPages2(currentPath);
-          Iterator<File> j = collPages.iterator();
-          boolean ok = false;
-          File f;
-          while (j.hasNext()) {
-            f = j.next();
-            if (f.getName().equals(nomPage)) {
-              ok = true;
-              break;
-            }
-          }
+    String destination = "/webSites/jsp/modifDesc.jsp?Id=" + id;
+    if (complete != null) {
+      destination += complete;
+    }
+    if (currentPath != null) {
+      destination += FOLDER_PATH_FILTER + currentPath;
+    }
+    if (type != null) {
+      destination += "&type=" + type;
+    }
+    return destination;
+  }
 
-          boolean searchOk = ok;
+  private String moveDownPublication(final WebSiteSessionController scc, final HttpRequest request)
+      throws WebSitesException {
+    String pubId = request.getParameter("Id");
 
-          SiteDetail descriptionSite2 =
-              new SiteDetail(id, scc.getComponentId(), nomSite, description, nomPage, type, null,
-                  null, Integer.parseInt(etat), popup);
+    scc.changePubsOrder(pubId, 1);
 
-          if (searchOk) {
+    String id = scc.getSessionTopic().getNodePK().getId();
+    FolderDetail webSitesCurrentFolder = scc.getFolder(id);
+    scc.setSessionTopic(webSitesCurrentFolder);
+    request.setAttribute(CURRENT_FOLDER_PARAM, webSitesCurrentFolder);
 
-            /* update description en BD */
-            scc.updateWebSite(descriptionSite2);
+    return "/webSites/jsp/organize.jsp?Action=Search&Id=" + id;
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "/webSites/jsp/organize.jsp?Action=Search&Id=" 4 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS1192)